src/components/dialogs/Dialog.js

```diff
@@ -0,0 +1,127 @@
+import Toggleable from '../../mixins/toggleable'
+
+export default {
+  name: 'dialog',
+
+  mixins: [Toggleable],
+
+  data () {
+    return {
+      stackedActions: false
+    }
+  },
+
+  props: {
+    persistent: Boolean,
+    fullscreen: Boolean,
+    overlay: Boolean,
+    removeTransition: Boolean,
+    origin: {
+      type: String,
+      default: 'center center'
+    },
+    transition: {
+      type: String,
+      default: 'v-modal-transition'
+    }
+  },
+
+  computed: {
+    classes () {
+      return {
+        'dialog': true,
+        'dialog--active': this.isActive,
+        'dialog--persistent': this.persistent,
+        'dialog--fullscreen': this.fullscreen,
+        'dialog--stacked-actions': this.stackedActions && !this.fullscreen,
+      }
+    },
+
+    computedOrigin () {
+      return this.origin
+    },
+
+    computedTransition () {
+      return this.transition
+    },
+
+    overlayClasses () {
+      return {
+        'overlay--modal-bottom': false  // this.bottom
+      }
+    }
+  },
+
+  methods: {
+    wasClickInside (target) {
+      return this.$refs.dialog !== target && !this.$refs.dialog.contains(target)
+    },
+
+    closeConditional (e) {
+      return this.persistent ? false : this.wasClickInside()
+    },
+
+    resize () {
+      if (!this.isActive) return
+
+      let actions = this.$children.filter(c => c.$options.propsData.actions )
+
+      // make sure we have the actions card row
+      if (actions.length) {
+        let btns = actions[0].$slots.default
+
+        let maxButtonWidth = (this.$slots.default[0].elm.offsetWidth - 8 - (8 * btns.length) ) / btns.length
```

Comment:
I'm a fan of using refs over slots as the instance you receive from slots as I've had mixed behavior.